### DISCUSSION TITLE
Update Policy changes

### DISCUSSION THREAD

**Discussion by olkornii** [eclipse-m2e/m2e-core#1542] [Q&A] [answered]
Hi everyone,

I have a question about Global Update Policy updates: https://github.com/eclipse-m2e/m2e-core/pull/1419
There used to be a checkbox, now it's a dropdown menu.

As far as I understand from the PR, previously CHECKED checkbox means "Never" in the new dropdown and UNCHECKED means "Default" in the new dropdown. So, if I want the same state as the previously CHECKED checkbox, I should choose "Never" in dropdown menu, then if I want the same state as the UNCHECKED checkbox before, then I should choose "Default", am I right?

@laeubi sorry to pinging you, but I think you have the most accurate information about that. I have limited time so a quick reply would be greatly appreciated.

**Comment by laeubi** [comment]
Default means some kind of default (e.g. using what ever is the default in maven that currently check every 24 hrs).

So you should simply choose what you desire here, the checkbox was a bit confusing as different settings where probably used across different functions and this is now similar to what is [described](https://maven.apache.org/settings.html#repositories) here
 under `updatePolicy`.

**Reply by olkornii** [reply]
@laeubi 
Hi, thanks for your reply, but I still don't fully understand.

I will try to explain from the other side. I'm working on testing framework, after this update I have to change the code and stay backwards compatible. So I need to know what "Checked" and "Unchecked" checkbox means to set the same options in the new dropdown.

In other words, I'd like to know what currently matches the checked and unchecked checkboxes. What values of dropdown menu.

**Reply by olkornii** [reply]
@laeubi 
As far as I understand now, the checked checkbox is the same as "Never" in the dropdown, unchecked == "Default" in the dropdown. Is it correct?

**Reply by olkornii** [reply]
@laeubi 
I have about 5-6 hours left to figure it out. I will be very grateful for a quick response.

**Reply by laeubi** [reply]
There is no backward compatible way for this, as the meaning is different, you might assume that checked is equal to never.
Beside that ist a bit strange that a testing framework depends on the m2e UI state... so you probably better want to check the preferences directly where you can of course set older values that the might be ignored by m2e.

**Reply by olkornii** [reply]
@laeubi 
ok, so checked is "never", then what is unchecked? "Default" ?

**Reply by laeubi** [reply]
Unchecked has no equivalent.

**Reply by olkornii** [reply]
In your Commit you have "Currently one can only select that updates are never performed or some
"defaults" are used.". Does "defaults" means something else? I thought it is "default".

**Reply by laeubi** [reply, marked as answer]
`Never` before mean never as in that it was explicitly disabled.  `some "defaults" ` before means that something unspecified was used.

Now `Never` means still never, and `Default` means using the default value that maven would use (e.g. from the pom, settings, commandline options, ...)

**Reply by olkornii** [reply]
@laeubi 
I see. Thank you. Now I know what to do :)
